src/App.tsx

```diff
@@ -67,7 +67,7 @@ export function App() {
   }, [darkMode, resolvedLanguage]);
 
   useEffect(() => {
-    if (user) {
+    if (user && Object.keys(user).length) {
```

Comment:
What exactly are we checking here for?

AFAIK `user` is `User | undefined` the first part of the check should be enough.

Reply:
I agree with your point, but actually when we enter the application the initial value of the user is an empty object {} but it is not undefined. An empty object is also a truthy value, so just putting that part in the if statement isn't enough. The second part is just checking if the object is empty, if it is, we won't do anything. This was one of the most important parts for making bug on the production.

Reply:
Got it, I thought it was undefined or populated object. Thanks!